### PR TITLE
Elemental analysis bugs

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**
Elemental analysis had several bugs:
1. Did not work as it had Python 2 only code
2. The co-add did not work
3. The plotting did not work

**To test:**
Open elemental analysis
Tick one of the detectors
Load some data
A plot should appear
Load some different data
The plot should be replaced

Select some data and press co-add instead of load (e.g. 2695)
Change the run number and press co-add again (e.g. 2696)
You should have some co-added runs in the ADS (2695-2696)
If you tick the detector a plot should appear

No associated issue

In release notes
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
@Mantid-Matthew I think this is ready for review now

File: scripts/test/Muon/LoadWidgetPresenter_test.py [hunk 1]
```diff
@@ -24,11 +24,19 @@ def setUp(self):
...
-        self.assertEqual(self.presenter.co_model.loaded_runs, 10)
+    def test_equalise_last_loaded_run_empty(self):
+        runs = OrderedDict()
+        self.presenter.co_model.loaded_runs = runs
+        self.presenter.load_model.loaded_runs = runs
+        self.presenter.equalise_last_loaded_run(runs)
+        self.assertEqual(self.presenter._current_run, None)
+
+    def test_equalise_last_loaded_run_data(self):
+        runs = OrderedDict({1:[], 2:[], 5:[], 3:[]})
```
Comment (AnthonyLim23) [inline scripts/test/Muon/LoadWidgetPresenter_test.py]:
```suggestion
        # need to add each in tern for Python 2
        runs = OrderedDict()
        runs[1]=[]
        runs[2] =[]
        runs[5] =[]
        runs[3]=[]
```

File: scripts/test/Muon/LoadWidgetPresenter_test.py [hunk 2]
```diff
@@ -24,11 +24,24 @@ def setUp(self):
...
+        # need to add each in tern for Python 2
+        runs = OrderedDict()
+        runs[1]=[]
+        runs[2] =[]
+        runs[5] =[]
+        runs[3]=[]
+        self.presenter.co_model.loaded_runs = runs
+        self.presenter.load_model.loaded_runs = runs
+        self.presenter.equalise_last_loaded_run(runs)
+        self.assertEqual(self.presenter._current_run, 3)
```
Comment (AnthonyLim23) [inline scripts/test/Muon/LoadWidgetPresenter_test.py]:
```suggestion
        self.assertEqual(self.presenter._current_run, '3')
```

File: scripts/Muon/GUI/Common/load_widget/load_presenter.py [hunk 3]
```diff
@@ -88,4 +89,10 @@ def unreg_on_loading_finished(self, slot):
         self.view.unreg_on_loading_finished(slot)
 
     def get_run_num_loaded_detectors(self, run):
-        return self.load_model.num_loaded_detectors[run]
+        num_detectors = 0
+        try:
+            group = mantid.mtd[run]
+            num_detectors = len(group)
+        except:
```
Comment (Mantid-Matthew) [inline scripts/Muon/GUI/Common/load_widget/load_presenter.py]: Is it possible to have a more specific exception here ?

**Review by Matthew-Andrew — COMMENTED** [review]
Can now load and display Elemental analysis data as expected. One Question below but otherwise code looks good as well.

File: scripts/Muon/GUI/Common/load_widget/load_presenter.py (hunk 3 above)
Comment (AnthonyLim23) [reply scripts/Muon/GUI/Common/load_widget/load_presenter.py]: Is this a `TypeError`?

Comment (AnthonyLim23) [reply scripts/Muon/GUI/Common/load_widget/load_presenter.py]:
```suggestion
        except KeyError:
```

**Review by Matthew-Andrew — APPROVED** [review]
Good to go

**Pull request merged**